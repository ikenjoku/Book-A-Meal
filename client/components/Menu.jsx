import React, { Component } from 'react';
import PropTypes from 'prop-types';

import NavBar from './NavBar.jsx';

import { menuSample } from "./mocks";

export default class Menu extends Component {
  constructor(props){
    super(props)
  }

  render(){
    return (
    <div>
      <NavBar/>
      <main className="menu-list-content">
        <div className="menu-list">
          <MenuTable/>
        </div>
        <div className="order-summary">
          <div className="order-summary-wrap">
            <PreviousOrderTable/>
            <CurrentOrderTable/>
          </div>
        </div>
      </main>
      <footer className="footer">
          <p>Book-A-Meal &copy;2018</p>
      </footer>     
    </div>
    );
  }
}

class MenuTable extends Component{
  constructor(props){
    super(props);
    this.state = {
      meals: menuSample.Meals
    }
  }

  render(){
    return (
      <div>
      <h2 className="center cool-lg-text">Menu List</h2>
        <MenuSearch/>
        <MealList meals={this.state.meals}/>
      </div>
    );
  }
}

class MenuSearch extends Component{
  constructor(props){
    super(props);
  }

  render(){
    return (
      <div>
        <div className="date-picker-pad order-history-date-picker center">
          <form>
            <input 
              id="datePicker" 
              className="input-field-control" 
              type="date" 
              placeholder="Choose a date..." 
              name="date"/>
            <input 
              id="datePickerBtn" 
              className="input-btn-control" 
              type="submit" 
              value="Choose date"/>
          </form>
        </div>
      </div>
    );
  }
}

const MealList = ({meals}) => {

  return (
    <div>
      {meals.map(meal =>
        <div key={meal.id} className="food-item">
          <img src={ require("../static/mealImg.jpg") } alt="" />
          <div className="food-title-desc">
              <p className="food-item-title">{ meal.name }</p>
              <p className="food-item-detail">{ meal.description }</p>
          </div>
          <div className="food-item-price">
              { meal.price }
          </div>
          <div><button className="food-add-btn">Order</button></div>
      </div> 
      )}
    </div>
  );
}

const PreviousOrderTable = (props) => {

  return (
    <div>
    <h4>Previous orders</h4>
    <div className="order-item">
        <div className="order-item-qty">07/3/17</div>
        <div className="order-item-name">Utazi</div>
        <div className="order-item-price">N1400 </div>
    </div>
    <div className="order-item">
        <div className="order-item-qty">12/4/17</div>
        <div className="order-item-name">Jollof rice</div>
        <div className="order-item-price">N200 </div>
    </div>
    <div className="order-item">
        <div className="order-item-qty">15/3/18</div>
        <div className="order-item-name">Boiled Egg</div>
        <div className="order-item-price">N100 </div>
    </div>
    </div>
  );
}


const CurrentOrderTable = (props) => {
  return (
    <div>
    <h4>Current Order</h4>
      <div className="order-item">
        <div className="order-item-qty">Egusi Soup</div>
        <div className="order-item-name">N1200</div>
        <div className="order-item-price"><button className="cancel-order-btn">Cancel Order</button> </div>
      </div>
      <hr/>
    </div>
  );
}
