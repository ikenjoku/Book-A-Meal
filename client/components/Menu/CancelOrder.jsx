import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { cancelAnOrder } from '../../actions/orderActions';

export const CancelOrder = ({ orderId, closeModal, cancelAnOrder }) =>
  (
    <div className="confirm-modal">
      <h3>Are you sure you want to cancel this order?</h3>
      <div>
        <button
          id="confirm-cancel-order"
          onClick={() => {
          cancelAnOrder(orderId);
          closeModal();
          }}
        >Yes
        </button>
        <button onClick={closeModal}>No</button>
      </div>
    </div>
  );

CancelOrder.propTypes = {
  orderId: PropTypes.number.isRequired,
  closeModal: PropTypes.func.isRequired,
  cancelAnOrder: PropTypes.func.isRequired,
};

export default connect(undefined, { cancelAnOrder })(CancelOrder);
