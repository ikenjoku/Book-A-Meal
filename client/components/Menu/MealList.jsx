import React from 'react';
import fallbackImage from "../../static/mealImg.jpg"


const MealList = ({ meals }) => (
  <div>
    {meals.map(meal =>
      <div key={meal.id} className="food-item">
        <img src={fallbackImage} alt="" />
        <div className="food-title-desc">
          <p className="food-item-title">{meal.name}</p>
          <p className="food-item-detail">{meal.description}</p>
        </div>
        <div className="food-item-price">
          {meal.price}
        </div>
        <div><button className="food-add-btn">Order</button></div>
      </div>
    )}
  </div>
);



export default MealList;