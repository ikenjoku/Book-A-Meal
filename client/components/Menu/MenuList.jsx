import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import modalStyle from '../../utils/modalStyle';
import PlaceOrder from './PlaceOrder';
import { orderAMeal } from '../../actions/orderActions';

export class MenuList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      mealId: undefined,
    };
  }

  handleOrder = (order) => {
    this.props.orderAMeal(order);
  }

  handleOpenModal = ({ mealId }) => {
    this.setState(() => ({ mealId, isOpen: true }));
  }

  handleCloseModal = () => {
    this.setState(() => ({ isOpen: false }));
  }

  render() {
    const { menu } = this.props;
    return (
      <div>
        {menu.Meals.length > 0 ?
          menu.Meals.map(meal =>
            (
              <div key={meal.id} className="food-item">
                <div className="meal-table-item-img">
                  <img src={meal.imageurl} alt="" />
                </div>
                <div className="food-title-desc">
                  <p className="food-item-title">{meal.name}</p>
                  <p className="food-item-detail">{meal.description}</p>
                </div>
                <div className="food-item-price">
                  &#8358; {meal.price}
                </div>
                <div>
                  {menu.date === new Date().toISOString().substr(0, 10) ?
                    <button
                      id={`order-meal-${meal.id}`}
                      className="food-add-btn"
                      onClick={() => { this.handleOpenModal({ mealId: meal.id }); }}
                    >
                      Place Order
                    </button> :
                    <p className="order-btn-text">Place order from todays menu</p>}
                </div>
              </div>
            )) :
          <p className="error-alert">No meals has been added to this menu.</p>}
        <Modal
          isOpen={this.state.isOpen}
          contentLabel="place-order"
          style={modalStyle}
          ariaHideApp={false}
        >
          <div className="close-icon">
            <button
              onClick={this.handleCloseModal}
            >
              <i className="fas fa-times fa-2x" />
            </button>
          </div>
          <PlaceOrder
            closeModal={this.handleCloseModal}
            mealId={this.state.mealId}
            handleOrder={this.handleOrder}
          />
        </Modal>
      </div>
    );
  }
}

MenuList.propTypes = {
  menu: PropTypes.object.isRequired,
  orderAMeal: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  menu: state.menuReducer.selectedMenu,
  isLoadingMenu: state.menuReducer.isLoading,
});

export default connect(mapStateToProps, { orderAMeal })(MenuList);
