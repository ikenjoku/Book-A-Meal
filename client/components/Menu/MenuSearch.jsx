import React, {Component} from 'react';

export default class MenuSearch extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        <div className="date-picker-pad order-history-date-picker center">
          <form>
            <input
              id="datePicker"
              className="input-field-control"
              type="date"
              placeholder="Choose a date..."
              name="date" />
            <input
              id="datePickerBtn"
              className="input-btn-control"
              type="submit"
              value="Choose date" />
          </form>
        </div>
      </div>
    );
  }
}
