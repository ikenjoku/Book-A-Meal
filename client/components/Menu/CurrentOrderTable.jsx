import React, { Component } from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import PropTypes from 'prop-types';
import modalStyle from '../../utils/modalStyle';
import CancelOrder from './CancelOrder';
import UpdateOrder from './UpdateOrder';
import { getAllPreviousOrders } from '../../actions/orderActions';

export class CurrentOrderTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      openCancelModal: false,
      openUpdateModal: false,
      orderId: undefined,
      orderToUpdate: undefined,
    };
  }

  componentDidMount() {
    this.props.getAllPreviousOrders(this.props.userId);
  }

  openCancelModal = ({ orderId }) => {
    this.setState(() => ({ orderId, openCancelModal: true }));
  }

  closeCancelModal = () => {
    this.setState(() => ({ openCancelModal: false }));
  }

  handleOrderUpdate = (orderToUpdate) => {
    this.setState(() => ({ orderToUpdate, openUpdateModal: true }));
  }

  closeUpdateModal = () => {
    this.setState(() => ({ openUpdateModal: false }));
  }

  render() {
    const { currentOrders } = this.props;
    const pendingOrders = currentOrders.filter(order => order.status === 'pending');

    return (
      <div>
        <h4>Current Order</h4>
        {pendingOrders.length === 0 ? <p>No current orders</p> :
          pendingOrders.map(order =>
            (
              <div key={order.id} className="order-item">
                <div className="order-item-name">{order.Meal.name}</div>
                <div className="order-item-amount">&#8358; {order.amount}</div>
                <div className="order-item-actions">
                  <button
                    className="update-order-icon"
                    onClick={() => { this.handleOrderUpdate(order); }}
                  >
                    <i className="far fa-edit" />
                  </button>
                  <button
                    className="cancel-order-icon"
                    onClick={() => {
                      this.openCancelModal({ orderId: order.id });
                    }}
                  >
                    <i className="far fa-times-circle 5x" />
                  </button>
                </div>
              </div>
            ))
          }
        <hr />
        <Modal
          isOpen={this.state.openCancelModal}
          contentLabel="cancel-order"
          style={modalStyle}
          ariaHideApp={false}
        >
          <div className="close-icon">
            <button
              onClick={this.closeCancelModal}
            >
              <i className="fas fa-times fa-2x" />
            </button>
          </div>
          <CancelOrder
            closeModal={this.closeCancelModal}
            orderId={this.state.orderId}
          />
        </Modal>
        <Modal
          isOpen={this.state.openUpdateModal}
          contentLabel="update-order"
          style={modalStyle}
          ariaHideApp={false}
        >
          <div className="close-icon">
            <button
              onClick={this.closeUpdateModal}
            >
              <i className="fas fa-times fa-2x" />
            </button>
          </div>
          <UpdateOrder
            closeModal={this.closeUpdateModal}
            orderToUpdate={this.state.orderToUpdate}
          />
        </Modal>
      </div>
    );
  }
}

CurrentOrderTable.propTypes = {
  currentOrders: PropTypes.arrayOf(PropTypes.object).isRequired,
  userId: PropTypes.number.isRequired,
  getAllPreviousOrders: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  currentOrders: state.orderReducer.previousOrders,
  userId: state.authReducer.user.id,
});

export default connect(mapStateToProps, { getAllPreviousOrders })(CurrentOrderTable);
