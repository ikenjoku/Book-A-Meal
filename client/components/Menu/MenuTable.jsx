import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MenuList from './MenuList';
import DatePicker from '../DatePicker';
import { getAMenu } from '../../actions/menuActions';

export class MenuTable extends Component {
  componentDidMount() {
    const today = new Date().toISOString().substr(0, 10);
    this.props.getAMenu({ selectedDate: today });
  }

  render() {
    const { menu } = this.props;
    return (
      <div>
        <h2 className="center cool-lg-text">Menu</h2>
        <DatePicker
          btnName="Get Menu"
          onSubmit={
          selectedDate => this.props.getAMenu(selectedDate)}
        />
        {this.props.error && <p className="error-alert">No Menu is set for this day</p>}
        {menu && !this.props.error ? <MenuList /> : ''}
      </div>
    );
  }
}

MenuTable.propTypes = {
  getAMenu: PropTypes.func.isRequired,
  error: PropTypes.object,
  menu: PropTypes.object,
};

MenuTable.defaultProps = {
  error: null,
  menu: null,
};

export const mapStateToProps = state => ({
  menu: state.menuReducer.selectedMenu,
  error: state.menuReducer.error,
});

export default connect(mapStateToProps, { getAMenu })(MenuTable);
