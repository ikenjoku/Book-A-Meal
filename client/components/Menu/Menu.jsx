import React from 'react';
import MenuTable from './MenuTable';
import PreviousOrderTable from './PreviousOrderTable';
import CurrentOrderTable from './CurrentOrderTable';

const Menu = () =>
  (
    <main id="menu-page" className="menu-list-content">
      <div className="menu-list">
        <MenuTable />
      </div>
      <div className="order-summary">
        <div className="order-summary-wrap">
          <CurrentOrderTable />
          <PreviousOrderTable />
        </div>
      </div>
    </main>
  );

export default Menu;

