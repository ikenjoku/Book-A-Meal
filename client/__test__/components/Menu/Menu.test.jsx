import React from 'react';
import { shallow } from 'enzyme';
import Menu from '../../../components/Menu/Menu';

describe('Menu: Menu Component', () => {
  let wrapper;
  test('Should render Menu Component corectly', () => {
    wrapper = shallow(<Menu />);
    expect(wrapper).toMatchSnapshot();
  });
});
