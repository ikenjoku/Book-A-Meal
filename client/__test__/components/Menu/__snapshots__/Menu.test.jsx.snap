// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu: Menu Component Should render Menu Component corectly 1`] = `
<main
  className="menu-list-content"
  id="menu-page"
>
  <div
    className="menu-list"
  >
    <Connect(MenuTable) />
  </div>
  <div
    className="order-summary"
  >
    <div
      className="order-summary-wrap"
    >
      <Connect(CurrentOrderTable) />
      <Connect(PreviousOrderTable) />
    </div>
  </div>
</main>
`;
